Fast and Flexible GPU Quantization for Transformers.md

```diff
@@ -1,82 +1,193 @@
 
 The corresponding dequantization equation is even simpler:
 
-$$\tilde{r}=S(Q(r)+Z$$
+$$\tilde{r}=S Q(r)$$
 
-Some terminology: Formula (1) describes **uniform quantization** (since the quantized values are uniformly distribute over the input space).
-
-To calculate $S$ we select a **clipping range** $[\alpha, \beta]$ and then do:
+This method is called **uniform quantization** since the quantized values are uniformly distributed over the input space. To calculate $S$ we select a **clipping range** $[\alpha, \beta]$ and then use:
 
 $$S=\frac{\beta-\alpha}{2^b-1}$$
 
-where $b$ is the number of bits in our quantization scheme. An extra constraint we choose to enforce is $\alpha=-beta$, which is called **symmetric quantization**. Its benefit is that this is equivalent to setting $Z=0$, which simplifies the (de)quantization function.
+Here, $b$ is the number of bits in our quantization scheme. We choose to enforce $\alpha=-\beta$, which is known as **symmetric quantization**. This simplifies the (de)quantization function by setting $Z=0$.
 
 ### Dynamic vs Static Quantization
-A key question is how to determine the clipping range. Too small, and we’ll excessively “truncate” activations & weights. Too big, and we’ll lose precision.
+A key question is how to determine the clipping range. Too small, and we’ll excessively “truncate” activations and weights. Too big, and we’ll lose precision.
 
-Whilst model parameters can always be quantized offline, its activations can be quantized **dynamically** (the clipping range is calculated for each activation, during a forward pass) or **statically** (also offline). To do static quantization, we run some forward passes and measure the distribution of each activation in the network in order to calculate the clipping range. This process is called **calibration**. For more details, see the next section.
+While model parameters can always be quantized offline, its activations can be quantized **dynamically** (with the clipping range calculated for each activation during a forward pass) or **statically** (also offline). Static quantization involves running some forward passes, measuring the distribution of each activation in the network, and calculating the clipping range. This process is called **calibration**. For more information, see the next section.
 
-Dynamic quantization is generally more accurate but incurs a computational overhead when calibrating the scalar online. Therefore, **we** **only consider static quantization on GPU** as the scalar reduction is expensive relative to the INT8 matmul, and so can severely limit any performance gains.
+Dynamic quantization tends to be more accurate but requires additional computational overhead for online scalar calibration. As a result, **we only consider static quantization on GPU** because scalar reduction, relative to an INT8 matmul, can be costly and limit performance gains.
```

Comment:
```suggestion
Dynamic quantization tends to be more accurate but requires additional computational overhead for online scalar calibration. As a result, **we only consider static quantization on GPU** because scalar reduction (relative to an INT8 matmul) can be costly and limit performance gains.
```